packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -12,12 +12,187 @@ export interface CodeBuildStartBuildProps extends sfn.TaskStateBaseProps {
+   */
+  readonly imagePullCredentialsTypeOverride?: 'CODEBUILD' | 'SERVICE_ROLE';
+  /**
+   * Enable this flag to override the insecure SSL setting that is specified in the build project.
+   *
+   * @default - the one specified in the build project only if the build's source is GitHub Enterprise.
+   */
+  readonly insecureSslOverride?: boolean;
+  /**
+   * Log settings for this build.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly logsConfigOverride?: codebuild.CfnProject.LogsConfigProperty;
+  /**
+   * Enable this flag to override privileged mode in the build project.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly privilegedModeOverride?: boolean;
+  /**
+   * The number of minutes a build is allowed to be queued before it times out.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly queuedTimeoutInMinutesOverride?: number;
+  /**
+   * The credentials for access to a private registry.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly registryCredentialOverride?: codebuild.CfnProject.RegistryCredentialProperty;
+  /**
+   * Set to true to report to your source provider the status of a build's start and completion.
+   *
+   * @default - the one specified in the build project only if the build's source is GitHub, GitHub Enterprise, or Bitbucket.
+   */
+  readonly reportBuildStatusOverride?: boolean;
+  /**
+   * An array of ProjectArtifacts objects.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly secondaryArtifactsOverride?: [codebuild.IArtifacts];
+  /**
+   * An array of ProjectSource objects.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly secondarySourcesOverride?: [codebuild.Source];
+  /**
+   * An array of ProjectSourceVersion objects that specify one or more versions of the project's secondary sources to be used for this build only.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly secondarySourcesVersionOverride?: [codebuild.CfnProject.ProjectSourceVersionProperty];
+  /**
+   * The name of a service role for this build.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly serviceRoleOverride?: string;
+  /**
+   * An authorization type for this build.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly sourceAuthOverride?: codebuild.CfnProject.SourceAuthProperty;
+  /**
+   * The source location for the one defined in the build project.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly sourceLocationOverride?: string;
+  /**
+   * The source input defined in the build project.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly sourceTypeOverride?: 'CODECOMMIT' | 'CODEPIPELINE' | 'GITHUB' | 'S3' | 'BITBUCKET' | 'GITHUB_ENTERPRISE' | 'NO_SOURCE';
```

Comment:
are source types not exposed through the codebuild module through an enum/struct?